concrete/src/Routing/RoutingServiceProvider.php

```diff
@@ -10,7 +10,7 @@ class RoutingServiceProvider extends Provider
      */
     public function register()
     {
-        $this->app->singleton(RouterInterface::class);
+        $this->app->singleton(Router::class);
         $this->app->bind(RouterInterface::class, Router::class);
```

Comment:
What about something like this?
```php
$this->app->singleton(Router::class);
$this->app->alias(Router::class, RouterInterface::class);
```

IMHO `alias()` is much easy to understand than `bind()`, is a bit faster.

BTW, for both cases (`bind` vs `alias`) this is true:
```php
$instance1 =  $this->app->make(RouterInterface::class);
$instance2 =  $this->app->make(RouterInterface::class);
$instance3 =  $this->app->make(Router::class);
$instance4 =  $this->app->make(Router::class);
($instance1 === $instance2 && $instance2 === $instance3 && $instance3 === $instance4);
```